lambda_art.py

```diff
@@ -0,0 +1,190 @@
+"""TODO: Put your header comment here."""
+
+import random
+import math
+from PIL import Image
+
+
+def build_random_function(min_depth, max_depth):
+    """Build a random function.
+
+    Builds a random function of depth at least min_depth and depth at most
+    max_depth. (See the assignment write-up for the definition of depth
+    in this context)
+
+    Args:
+        min_depth: the minimum depth of the random function
+        max_depth: the maximum depth of the random function
+
+    Returns:
+        The randomly generated function represented as a nested list.
+        (See the assignment writ-eup for details on the representation of
+        these functions)
+    """
+
+    if min_depth < 1:
+        var = random.randint(1,2)
+        if var == 1:
+            return lambda x,y: x
```

Comment:
Your implementations of lambda functions are correct/functional, but you aren't quite realizing some of the benefits of lambda-izing - for example, lambda-izing functions allows you to choose/pass around functions instead of specifically checking things like rand_int values.

That  being said, I do quite like how you've used them to make build_random_function also build the evaluation path - that's quite elegant, in my opinion! All in all, solidly done.